### PR TITLE
add optional mkdocstrings-python dependency

### PR DESCRIPTION
**Pull request by bandersen23**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -32,6 +32,7 @@ requirements:
     - pymdown-extensions >=6.3
   run_constrained:
     - mkdocstrings-python-legacy >=0.2
+    - mkdocstrings-python >=5.2
```
Comment (hadim) [inline recipe/meta.yaml:35]: What do you think about having `mkdocstrings-python-legacy` and `mkdocstrings-python` as hard deps (once the conda feedstock has finally been created ofc)?

Pretty sure we will have more usecase where people want to have handlers than the opposite. That would also keep the previous behaviour where all the deps are within `mkdocstring`.

ping @bandersen23

Comment (bandersen23) [reply recipe/meta.yaml:35]: That's a good idea, I'll update it in #24 when the feedstocks are all officially created.

Comment (hadim) [reply recipe/meta.yaml:35]: Thanks.

Comment (bandersen23) [reply recipe/meta.yaml:35]: Done in ba88b67 thru #24.

**Pull request merged**